mmcv/runner/base_module.py

```diff
@@ -38,12 +63,38 @@ def is_init(self):
 
     def init_weights(self):
         """Initialize the weights."""
-        from ..cnn import initialize
 
+        # judge if it is Topmost module
+        top_most_module = len(self.params_init_info) == 0
+        if top_most_module:
+            for name, param in self.named_parameters():
+                self.params_init_info[param]['params_name'] = name
+                self.params_init_info[param][
+                    'init_info'] = f'The value is the same before and ' \
+                                   f'after calling `init_weights` ' \
+                                   f'of {self.__class__.__name__} '
+                self.params_init_info[param]['tmp_sum_value'] = param.data.sum(
+                )
+            # pass `params_init_info` to all submodules
+            # all submodules will modify the same `params_init_info` \
+            # during initialization thus params_init_info will
+            # keep the final initialization
+            # information of each module.
+            for sub_moduls in self.modules():
+                sub_moduls.params_init_info = self.params_init_info
+
+        loggernames = list(logger_initialized.keys())
+        loggername = loggernames[0] if len(loggernames) > 0 else 'mmcv'
+
+        from ..cnn import initialize
+        modulename = self.__class__.__name__
         if not self._is_init:
             if self.init_cfg:
+                print_log(
+                    f'initialize {modulename} with init_cfg {self.init_cfg}',
+                    logger=loggername)
```

Comment:
Have you checked if this information will be saved in the log file?

Reply:
I have checked it. It will, but getting the logger name (L87) is not elegant.